src/main/java/com/mongodb/jdbc/BuildInfo.java

```diff
@@ -16,13 +16,54 @@
 
 package com.mongodb.jdbc;
 
+import java.util.Optional;
 import java.util.Set;
-import org.bson.BsonValue;
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonProperty;
 
-// Simple POJO for deserializing buildInfo results.
 public class BuildInfo {
     public String version;
     public Set<String> modules;
-    public BsonValue dataLake;
     public int ok;
+    public DataLake dataLake;
+
+    @BsonCreator
```

Comment:
You don't need to use the annotations when there are no customization, it's a simple POJO.
See JsonSchema or MongoJsonSchema. And the flower example https://www.mongodb.com/docs/drivers/java/sync/v5.2/fundamentals/data-formats/document-data-format-pojo/

Now, if you want, you could bake in the logic for the "calculated" version in your constructor.
If there is a dataLake section, then add the datalake version and mongosql version. I would even make this another variable. Like fullVersion and use that directly in MongoConnection.
It will be cleaner/simpler.

Reply:
Yep going with this approach